Test/VsVimTest/ReferenceTest.cs

```diff
@@ -0,0 +1,305 @@
+                        return new ReferenceData(assemblyName, ReferenceKind.Editor, getVersionFromVersionNumber());
+
+                    case "Vim.Core":
+                    case "Vim.UI.Wpf":
+                    case "Vim.VisualStudio.Shared":
+                    case "Vim.VisualStudio.Interfaces":
+                        return new ReferenceData(assemblyName, ReferenceKind.VsVim, VsVersion.MinimumSupported);
+
+                    case "EnvDTE":
+                    case "EnvDTE80":
+                    case "EnvDTE90":
+                    case "EnvDTE100":
+                    case "Microsoft.VisualStudio.OLE.Interop":
+                    case "Microsoft.VisualStudio.Shell.Interop":
+                    case "Microsoft.VisualStudio.Shell.Interop.8.0":
+                    case "Microsoft.VisualStudio.Shell.Interop.9.0":
+                    case "Microsoft.VisualStudio.Shell.Interop.10.0":
+                    case "Microsoft.VisualStudio.Shell.Interop.11.0":
+                    case "Microsoft.VisualStudio.Shell.Interop.12.0":
+                    case "Microsoft.VisualStudio.TextManager.Interop":
+                    case "Microsoft.VisualStudio.TextManager.Interop.8.0":
+                    case "Microsoft.VisualStudio.TextManager.Interop.10.0":
+                        return new ReferenceData(assemblyName, ReferenceKind.ShellPia, VsVersion.MinimumSupported);
+
+                    case "Microsoft.VisualStudio.ComponentModelHost":
+                    case "Microsoft.VisualStudio.Editor":
+                    case "Microsoft.VisualStudio.Shell.10.0":
+                    case "Microsoft.VisualStudio.Shell.11.0":
+                    case "Microsoft.VisualStudio.Shell.12.0":
+                    case "Microsoft.VisualStudio.Shell.Immutable.10.0":
+                    case "Microsoft.VisualStudio.Shell.ViewManager":
+                    case "Microsoft.VisualStudio.Utilities":
+                        return new ReferenceData(assemblyName, ReferenceKind.ShellVersioned, getVersionFromVersionNumber());
+
+                    case "Microsoft.VisualStudio.Platform.WindowManagement":
+                        return new ReferenceData(assemblyName, ReferenceKind.ShellNonVersioned, getVersionFromVersionNumber());
+
+                    default:
+                        throw new Exception($"Unrecognized reference {name}");
+                }
+
+                VsVersion getVersionFromVersionNumber()
+                {
+                    var value = assemblyName.Version.Major;
+                    if (Enum.IsDefined(typeof(VsVersion), value))
+                    {
+                        return (VsVersion)value;
+                    }
+
+                    throw new InvalidOperationException($"Invalid enum value {value}");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Make sure the correct VS SDK binaries are referenced in the core binaries.

```

Comment:
Great that this is automated.

Reply:
Agree. The rules here are individually easy enough but there are so many to remember and it can be difficult to trace them through the build process. Automated so hopefully I can forget about this going forward 😄